Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp

```diff
@@ -4109,9 +4109,11 @@ void GraphicsLayerCA::updateRootRelativeScale()
     };
 
     float rootRelativeScaleFactor = hasNonIdentityTransform() ? computeMaxScaleFromTransform(transform()) : 1;
-
-    if (auto* parentLayer = parent(); parentLayer && parentLayer->hasNonIdentityChildrenTransform())
-        rootRelativeScaleFactor = std::max(rootRelativeScaleFactor, computeMaxScaleFromTransform(parentLayer->childrenTransform()));
+    if (m_parent) {
+        if (m_parent->hasNonIdentityChildrenTransform())
```

Comment:
We have to be careful here and only apply the logic for LBSE layers.

Reply:
It's already applied only to LBSE layers because we return early from `updateRootRelativeScale()` if `m_shouldUpdateRootRelativeScaleFactor` is not set:
https://github.com/WebKit/WebKit/blob/aa49cd4d27107bd400dbbffc2542ddd5e30ab8ed/Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp#L4092-L4101
And we set `m_shouldUpdateRootRelativeScaleFactor` only for SVG layers and only if LBSE is enabled:
https://github.com/WebKit/WebKit/blob/aa49cd4d27107bd400dbbffc2542ddd5e30ab8ed/Source/WebCore/rendering/RenderLayerBacking.cpp#L370-L371
For all other layers `m_rootRelativeScaleFactor` always stays equal to `1.0f`.